apis/v1alpha2/shared_types.go

```diff
@@ -249,4 +238,41 @@ type RouteStatus struct {
 //
 // +kubebuilder:validation:MinLength=1
 // +kubebuilder:validation:MaxLength=253
+// +kubebuilder:validation:Pattern=`^(\*.)?[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$`
 type Hostname string
+
+// Group refers to a Kubernetes Group.
+//
+// +kubebuilder:validation:MaxLength=253
+// +kubebuilder:validation:Pattern=`^$|^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$`
+type Group string
```

Comment:
I wish this was defined in upstream and we could simply pull it from there. Some other day. No action required.

Reply:
Do we have a reference as to where this regex is using as the basis for tis definition?

Reply:
I added a note for this in one of my favorite issues: https://github.com/kubernetes-sigs/gateway-api/issues/629